community/kernel/src/main/java/org/neo4j/kernel/impl/store/MetaDataStore.java

```diff
@@ -396,6 +404,13 @@ public void setCurrentLogVersion( long version )
         }
     }
 
+    public void setLastTransactionCommitTimestamp( long timestamp )
+    {
+        setRecord( Position.LAST_TRANSACTION_COMMIT_TIMESTAMP, timestamp );
+        TransactionId transactionId = highestCommittedTransaction.get();
+        highestCommittedTransaction.set( transactionId.transactionId(), transactionId.checksum(), timestamp );
```

Comment:
This method body should be in a `synchronized ( transactionCommittedLock ) { … }` block to avoid racing with `transactionCommitted()`. Otherwise we can have word tearing on the page, and also we are not guarding the internals of `HighestTransactionId`.